core/blockchain.go

```diff
@@ -2087,8 +2090,10 @@ func (bc *BlockChain) SetChainHead(head *types.Block) error {
 	// Run the reorg if necessary and set the given block as new head.
 	start := time.Now()
 	if head.ParentHash() != bc.CurrentBlock().Hash() {
-		if err := bc.reorg(bc.CurrentBlock(), head); err != nil {
-			return err
+		if head.Hash() != bc.CurrentBlock().Hash() {
```

Comment:
I don't follow this change? Seems that it's fixing the symptom instead of the root cause. You shouldn't call SetHead to the current head. Silently noop-ing isn't the correct behavior, the correct fix would be not to do the wrong call in the first place.